manager/runner.go

```diff
@@ -750,23 +733,55 @@ func (r *Runner) configTemplatesFor(tmpl *template.Template) []*config.ConfigTem
 	return r.ctemplatesMap[tmpl.ID()]
 }
 
+// ConfigTemplateMapping returns a mapping between the template ID and the set
+// of ConfigTemplate represented by the template ID
+func (r *Runner) ConfigTemplateMapping() map[string][]config.ConfigTemplate {
+	m := make(map[string][]config.ConfigTemplate, len(r.ctemplatesMap))
+
+	for id, set := range r.ctemplatesMap {
+		ctmpls := make([]config.ConfigTemplate, len(set))
+		m[id] = ctmpls
+		for i, ctmpl := range set {
+			ctmpls[i] = *ctmpl
+		}
+	}
+
+	return m
+}
+
 // allTemplatesRendered returns true if all the templates in this Runner have
 // been rendered at least one time.
 func (r *Runner) allTemplatesRendered() bool {
-	r.renderedTimesLock.RLock()
-	defer r.renderedTimesLock.RUnlock()
+	r.renderEventsLock.RLock()
+	defer r.renderEventsLock.RUnlock()
 
 	for _, tmpl := range r.templates {
-		for _, ctmpl := range r.configTemplatesFor(tmpl) {
-			if _, rendered := r.renderedTimes[ctmpl.Destination]; !rendered {
-				return false
-			}
+		if _, rendered := r.renderEvents[tmpl.ID()]; !rendered {
+			return false
 		}
 	}
 
 	return true
 }
 
+// markRenderTime stores the render time for the given template. If didRender is
+// true, it stores the time for the template having been rendered, otherwise it
+// stores it as would have been rendered.
+func (r *Runner) markRenderTime(tmplID string, didRender bool) {
+	r.renderEventsLock.Lock()
+	event, ok := r.renderEvents[tmplID]
+	if !ok {
+		event = &RenderEvent{}
+		r.renderEvents[tmplID] = event
+	}
+	if didRender {
+		event.LastDidRender = time.Now()
```

Comment:
This is super picky, but I think there's a tiny race here where these two `time.Now()`s are actually different times. Could we grab the current time right after we acquire the lock, just to be safe?